rpc/ws.ts

```diff
@@ -49,7 +49,28 @@ export class WsConnection extends Connection {
     this.ws.send(RpcEgressMessage.fmt(id, method, params))
   }
 
-  close() {
-    this.ws.close()
+  async close() {
+    switch (this.ws.readyState) {
+      case WebSocket.OPEN: {
+        this.ws.close()
+        break
+      }
+      case WebSocket.CONNECTING: {
+        const controller = new AbortController()
+        const options: AddEventListenerOptions = {
+          signal: controller.signal,
+          once: true,
+        }
+        this.ws.addEventListener("open", () => {
+          this.ws.close()
+          controller.abort()
+        }, options)
+        this.ws.addEventListener("close", () => {}, options)
```

Comment:
do we need this if the handle block is ermpty?

Reply:
No, this is a desirable outcome (as we're trying to close the socket anyways).